tests/30rooms/13guestaccess.pl

```diff
@@ -172,26 +172,28 @@
                displayname => "creeper",
             },
       )})->then( sub {
-         Future->needs_all(
-            do_request_json_for( $guest_user,
-               method => "GET",
-               uri    => $displayname_uri,
-            )->then( sub {
-               my ( $body ) = @_;
-               assert_eq( $body->{displayname}, "creeper", "Profile displayname" );
-
-               Future->done(1);
-            }),
-            do_request_json_for( $guest_user,
-               method => "GET",
-               uri    => "/r0/rooms/$room_id/state/m.room.member/:user_id",
-            )->then( sub {
-               my ( $body ) = @_;
-               assert_eq( $body->{displayname}, "creeper", "Room displayname" );
-
-               Future->done(1);
-            }),
-         );
+         repeat_until_true {
```

Comment:
Do we not want `retry_until_success`? As the functions are presumably throwing errors rather than return false.

Reply:
Good call. I think my brain made those into synonyms.